### PR TITLE
lib: lte_link_control: Remove CONFIG_LTE_NETWORK_USE_FALLBACK

### PR DESCRIPTION
**Pull request by trantanen**
`CONFIG_LTE_NETWORK_USE_FALLBACK` has been deprecated by PR #13280 and is now removed.

TODO:

- [x] release note
- [x] migration guide

### CODE REVIEW

File: lib/lte_link_control/lte_lc.c [hunk 1]
```diff
@@ -744,8 +702,7 @@ static int connect_lte(bool blocking)
 		goto exit;
 	}
 
-	if (lte_lc_sys_mode != LTE_LC_SYSTEM_MODE_DEFAULT &&
-	    IS_ENABLED(CONFIG_LTE_NETWORK_USE_FALLBACK)) {
+	if (lte_lc_sys_mode != LTE_LC_SYSTEM_MODE_DEFAULT) {
 		/* Set system mode, because it may have been changed by fallback. */
 		if (original_func_mode != LTE_LC_FUNC_MODE_POWER_OFF &&
 		    original_func_mode != LTE_LC_FUNC_MODE_OFFLINE) {
```
Comment (trantanen) [inline lib/lte_link_control/lte_lc.c:708]: @tokangas, Need to discuss what to do with this, i.e., what is correct behavior here.

Comment (trantanen) [reply lib/lte_link_control/lte_lc.c]: This whole branch has been removed after offline discussion.

File: tests/lib/lte_lc_api/src/lte_lc_api_test.c [hunk 2]
```diff
@@ -469,15 +374,10 @@ void test_lte_lc_connect_timeout(void)
...
-	__mock_nrf_modem_at_printf_ExpectAndReturn(
-		"AT%XSYSTEMMODE=1,1,0,0", 0);
-
 	__mock_nrf_modem_at_printf_ExpectAndReturn("AT+CEREG=5", 0);
 	__mock_nrf_modem_at_printf_ExpectAndReturn("AT+CSCON=1", 0);
 	__mock_nrf_modem_at_printf_ExpectAndReturn("AT+CFUN=1", 0);
 
-	/* Modem is switched into offline mode after the timeout */
-	__mock_nrf_modem_at_printf_ExpectAndReturn("AT+CFUN=4", 0);
 	/* Fallback not possible, modem switched back to original functional mode */
```
Comment (tokangas) [inline tests/lib/lte_lc_api/src/lte_lc_api_test.c:381]:
```suggestion
	/* Modem switched back to original functional mode after the timeout */
```

**Review by tokangas — APPROVED** [review]

**Review by tokangas — APPROVED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 3]
```diff
@@ -474,6 +474,9 @@ Modem libraries
       Use the :c:func:`lte_lc_connect` function instead.
     * The :c:func:`lte_lc_init_and_connect_async` function.
       Use the :c:func:`lte_lc_connect_async` function instead.
+    * The :kconfig:option:`CONFIG_LTE_NETWORK_USE_FALLBACK` Kconfig option.
```
Comment (wiba-nordic) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:477]:
```suggestion
    * The ``CONFIG_LTE_NETWORK_USE_FALLBACK`` Kconfig option.
```

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.8.rst [hunk 4]
```diff
@@ -51,6 +51,10 @@ LTE link control library
      * Replace the use of the :c:func:`lte_lc_deinit` function with the :c:func:`lte_lc_power_off` function.
      * Replace the use of the :c:func:`lte_lc_init_and_connect` function with the :c:func:`lte_lc_connect` function.
      * Replace the use of the :c:func:`lte_lc_init_and_connect_async` function with the :c:func:`lte_lc_connect_async` function.
+     * Remove the use of the :kconfig:option:`CONFIG_LTE_NETWORK_USE_FALLBACK` Kconfig option.
```
Comment (wiba-nordic) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.8.rst:54]:
```suggestion
     * Remove the use of the ``CONFIG_LTE_NETWORK_USE_FALLBACK`` Kconfig option.
```

**Review by wiba-nordic — CHANGES_REQUESTED** [review]
The deprecated Kconfig option is mentioned here in the documentation, so this section needs to be edited as well:

https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/libraries/modem/lte_lc.html#connection_fallback_mode

**Comment by trantanen** [comment]
> The deprecated Kconfig option is mentioned here in the documentation, so this section needs to be edited as well:
> 
> https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/libraries/modem/lte_lc.html#connection_fallback_mode

True, I'll check that tomorrow. Hold on with the review until then.

File: doc/nrf/libraries/modem/lte_lc.rst [hunk 5]
```diff
@@ -184,16 +184,17 @@ To enable modem sleep and TAU pre-warning notifications, enable the following op
...
+It is possible to try to switch between LTE-M and NB-IoT if a connection has not been established.
 This is useful when the connection to either of these networks becomes unavailable.
-You can also configure the switching period between the network modes.
-If a connection cannot be established by using the fallback mode, the library reports an error.
-You can use the following configuration options to configure the connection fallback mode:
 
-* :kconfig:option:`CONFIG_LTE_NETWORK_USE_FALLBACK`
-* :kconfig:option:`CONFIG_LTE_NETWORK_TIMEOUT`
+Use the :kconfig:option:`CONFIG_LTE_NETWORK_MODE_LTE_M_NBIOT` or :kconfig:option:`CONFIG_LTE_NETWORK_MODE_LTE_M_NBIOT_GPS` Kconfig option
+with which the modem will perform a fallback from one system mode to another if a connection to the first one fails.
```
Comment (wiba-nordic) [inline doc/nrf/libraries/modem/lte_lc.rst:193]:
```suggestion
Use the :kconfig:option:`CONFIG_LTE_NETWORK_MODE_LTE_M_NBIOT` or :kconfig:option:`CONFIG_LTE_NETWORK_MODE_LTE_M_NBIOT_GPS` Kconfig option to make the modem perform a fallback from one system mode to another if a connection to the first one fails.
```

File: doc/nrf/libraries/modem/lte_lc.rst [hunk 6]
```diff
@@ -184,16 +184,17 @@ To enable modem sleep and TAU pre-warning notifications, enable the following op
...
+
+In addition, you can control the priority between LTE-M and NB-IoT using the :kconfig:option:`CONFIG_LTE_MODE_PREFERENCE` Kconfig option.
+
+You can also configure the the overall connection establishment time with the :kconfig:option:`CONFIG_LTE_NETWORK_TIMEOUT` Kconfig option.
```
Comment (wiba-nordic) [inline doc/nrf/libraries/modem/lte_lc.rst:197]:
```suggestion
You can also configure the overall connection establishment time with the :kconfig:option:`CONFIG_LTE_NETWORK_TIMEOUT` Kconfig option.
```

**Review by wiba-nordic — CHANGES_REQUESTED** [review]

File: doc/nrf/libraries/modem/lte_lc.rst (hunk 5 above)
Comment (trantanen) [reply doc/nrf/libraries/modem/lte_lc.rst]: This is now removed after discussing with my colleagues

File: doc/nrf/libraries/modem/lte_lc.rst (hunk 6 above)
Comment (trantanen) [reply doc/nrf/libraries/modem/lte_lc.rst]: This is now removed after discussing with my colleagues

**Review by lemrey — APPROVED** [review]

**Review by wiba-nordic — APPROVED** [review]

**Pull request merged**